From 6458b82948a44416ad0a2315357544a4208f8bca Mon Sep 17 00:00:00 2001
From: "jaekyu.lee" <jaekyu.lee@lge.com>
Date: Thu, 23 Jan 2025 10:49:26 +0900
Subject: [PATCH] Fix build with liburcu-0.15.0

:Release Notes:

:Detailed Notes:
Fixes:
http://gecko.lge.com:8000/Errors/Details/1095224

compiling qwayland-qt-key-unstable-v1.cpp
In file included from TOPDIR/BUILD/work/o22-webos-linux/qtwayland/5.12.4-174/recipe-sysroot/usr/include/lttng/tracepoint-rcu.h:10,
                 from TOPDIR/BUILD/work/o22-webos-linux/qtwayland/5.12.4-174/recipe-sysroot/usr/include/lttng/tracepoint.h:13,
                 from TOPDIR/BUILD/work/o22-webos-linux/qtwayland/5.12.4-174/git/src/client/pmtrace_qtwaylandclient_provider.h:28,
                 from TOPDIR/BUILD/work/o22-webos-linux/qtwayland/5.12.4-174/git/src/client/qtwaylandclienttracer.h:14,
                 from TOPDIR/BUILD/work/o22-webos-linux/qtwayland/5.12.4-174/git/src/hardwareintegration/client/wayland-egl/qwaylandglcontext.cpp:49:
TOPDIR/BUILD/work/o22-webos-linux/qtwayland/5.12.4-174/recipe-sysroot/usr/include/urcu/compiler.h:205:1: error: template with C linkage
  205 | template <typename T>

:Testing Performed:
Only build tested.

:QA Notes:
No change to image.

:Issues Addressed:
[WRR-17603] Fix build with liburcu-0.15.0

Change-Id: I7467077afb1f7ab27bd75bd70426071c5b82fe32
---
Upstream-Status: Submitted [http://gpro.lge.com/c/webosose/qtwayland-webos/+/466467]

 .../pmtrace_qtwaylandwebos_provider.h              | 14 +++++++-------
 1 file changed, 7 insertions(+), 7 deletions(-)

diff --git a/src/plugins/platforms/webos-wayland-egl/pmtrace_qtwaylandwebos_provider.h b/src/plugins/platforms/webos-wayland-egl/pmtrace_qtwaylandwebos_provider.h
index 36d6aa2..913d412 100644
--- a/src/plugins/platforms/webos-wayland-egl/pmtrace_qtwaylandwebos_provider.h
+++ b/src/plugins/platforms/webos-wayland-egl/pmtrace_qtwaylandwebos_provider.h
@@ -20,16 +20,15 @@
 #undef TRACEPOINT_INCLUDE
 #define TRACEPOINT_INCLUDE "./pmtrace_qtwaylandwebos_provider.h"
 
-#ifdef __cplusplus
-extern "C"{
-#endif /*__cplusplus */
-
-
 #if !defined(_PMTRACE_QTWAYLANDWEBOS_PROVIDER_H) || defined(TRACEPOINT_HEADER_MULTI_READ)
 #define _PMTRACE_QTWAYLANDWEBOS_PROVIDER_H
 
 #include <lttng/tracepoint.h>
 
+#ifdef __cplusplus
+extern "C"{
+#endif /*__cplusplus */
+
 /* "message" tracepoint should be used for single event trace points */
 TRACEPOINT_EVENT(
     qtwayland_webos,
@@ -101,8 +100,9 @@ TRACEPOINT_EVENT(
     TP_ARGS(const char*, text),
     TP_FIELDS(ctf_string(scope, text)))
 
-#endif /* _PMTRACE_QTWAYLANDWEBOS_PROVIDER_H */
-
 #ifdef __cplusplus
 }
 #endif /*__cplusplus */
+
+#endif /* _PMTRACE_QTWAYLANDWEBOS_PROVIDER_H */
+
